lib/dialects/abstract/query-generator.js

```diff
@@ -1520,6 +1522,195 @@ module.exports = (function() {
       return result;
     },
 
+    whereQuery: function(where, options) {
+      var query = this.whereItemsQuery(where, options);
+      if (query && query.length) {
+        return 'WHERE '+query;
+      }
+      return '';
+    },
+    whereItemsQuery: function(where, options, binding) {
+      if (
+        (Array.isArray(where) && where.length === 0) ||
+        (_.isPlainObject(where) && _.isEmpty(where)) ||
+        where === '' ||
+        where === null ||
+        where === undefined
+      ) {
+        // NO OP
+        return '';
+      }
+
+      var self = this
+        , items = [];
+
+      binding = ' '+ (binding || 'AND') + ' ';
+
+      if (_.isPlainObject(where)) {
+        _.forIn(where, function (value, key) {
+          items.push(self.whereItemQuery(key, value, options));
+        });
+      } else {
+        items.push(self.whereItemQuery(undefined, where, options));
+      }
+
+      return items.length && items.filter(function (item) {
+        return item && item.length;
+      }).join(binding) || '';
+    },
+    whereItemQuery: function(key, value, options) {
+      var self = this
+        , binding
+        , comparatorMap
+        , aliasMap
+        , comparator = '=';
+
+      options = options || {};
+
+      comparatorMap = {
```

Comment:
mind you here its called "whereItemQuery" but you just mentioned "whereItem**s**Query". Also, what exactly does this mean? Why isn't it just whereQuery?